### DISCUSSION TITLE
Triggering `onDelete` programatically

### DISCUSSION THREAD

**Discussion by graddaniel** [xyflow/xyflow#4444] [Q&A]
I'd like to put whole the deletion related code (API calls, etc) into the `onDelete` handler, but unfortunately it's called only on Delete Key press (backspace by default). Filtering out the edges/nodes and setting them to the filtered out array, simply sets them and doesn't trigger the callback.
I've found out that there used to be `removeElements` function in some of the older versions.

Is there any way, possibly a function exposed by the ReactFlow API, which allows to remove an edge and trigger the `onDelete` in the v12?

**Comment by bcakmakoglu** [comment]
You can use the [`deleteElements`](https://reactflow.dev/api-reference/hooks/use-react-flow#delete-elements) function (returned by `useReactFlow`).

```ts
const { deleteElements } = useReactFlow()

deleteElements({ nodes: [{ id: '1' }], edges: [{ id: 'e1->2' }] })
```

**Reply by graddaniel** [reply]
I forgot to mention that initially I wanted to do that outside the ReactFlow component, which complains about missing Zustand provider, but in the end there's no longer need to do that.
Thanks for the help anyway, this does the job now.

**Reply by bcakmakoglu** [reply]
[You might wanna read up on that warning in the troubleshooting section.](https://reactflow.dev/learn/troubleshooting#warning-seems-like-you-have-not-used-zustand-provider-as-an-ancestor)

**Reply by graddaniel** [reply]
How could I miss it?
Thanks!
